ci: run Flux-Parsl integration tests

jameshcorbett opened the pull request:
This should probably wait for #51 to go in.

Changes the parsl install from the default pip version to master, and adds an integration dockerfile and test script for testing integrations---not sure if that's the best structure.

jameshcorbett commented:
I don't understand why the CI failed so if anyone could explain I would appreciate it 👍

dongahn commented on docker/integration/test.sh:
```diff
@@ -0,0 +1,7 @@
+
+flux start python3 -m pytest $(python3 -c \
+	"import parsl.tests.test_flux as tf; print(tf.__file__)") \
+	--config=local --tap-stream
```
For my own edification, what does `--config=local` do?

jameshcorbett replied:
That's a parsl flag they added to `pytest`, the way you set up Parsl is you define a `config`, and the flag is basically to specify that I don't need a special site-specific config.

dongahn commented:
> I don't understand why the CI failed so if anyone could explain I would appreciate it 👍

Looks like something is returning 1.

![Screen Shot 2021-08-06 at 1 27 26 PM](https://user-images.githubusercontent.com/862123/128567960-8288a4cf-f291-408b-915f-48d0f12ab8d8.png)

dongahn commented:
Could be intermittent. Let me restart.

dongahn reviewed (approved):
Elegant!

dongahn commented:
For GitHub action CI debugging,  https://github.com/marketplace/actions/debugging-with-tmate might come in handy.

Probably adding the following step would give you a terminal into the container.
```
- name: Setup tmate session
      uses: mxschmitt/action-tmate@v3
```

dongahn commented:
This might also be a question for @SteVwonder and @andre-merzky. Are we capturing the logs from the commands running inside our containers in our GitHub action infrastructure? This may come in handy when we need to debug -- beyond being able to say, hey non-zero return from the `CMD`.

SteVwonder commented:
> Are we capturing the logs from the commands running inside our containers in our GitHub action infrastructure?

Not sure why the logs aren't popping up in this check.  For all of the other CI runs that I've seen, if the docker container (and the processes in it) write to stdout/stderr, then it gets propagated to the GH actions log.  I'm tempted to say this is just a weird GH bug/failure for that particular run.

In #51 , I added a flux-specific script that runs to try and summarize any Flux-specific failures that occurred.  We could do the same for other components too if we want.  We just need to get input from the other components for how to do that for their testsuits.

mtitov commented on .github/workflows/ci.yml:
```diff
@@ -55,8 +59,12 @@ jobs:
+        docker run exaworks_sdk_integration:${{ matrix.dockerbase }}
+        ret_integration=$?
+        echo "integration: $ret_integration"
+        exit $ret_integration
```
this line not needed (we shouldn't force to exit in CI run)

mtitov replied:
sorry, missed that we actually use it this way..

mtitov commented on docker/integration/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+# install pytest-tap to allow Parsl's pytest scripts to output TAP
+RUN python3 -m pip install pytest-tap
+
+COPY test.sh /
+CMD /test.sh
```
I would propose to use a different name (with the current name it will overwrite Parsl's individual test script) and maybe to put it in a dedicated directory, e.g., `/sdk_tests/` (later need to do the same with other test scripts)

dongahn commented:
The same here. This PR has been pending for awhile. I would say we discuss to see if we can unstick the current roadblock.

dongahn commented:
@James will address Mikhail's comment above.
@andre-merzky will look at CI issues above,

jameshcorbett commented:
CI passed! I'm now squashing all the incremental commits down to one. The CI should pass again, or I'll lose my mind.

dongahn commented:
Wow! heroic @jameshcorbett... you should tell me what the issue was later :-)

jameshcorbett commented:
Still a little bit of a mystery what went wrong. The only thing I changed was that I removed the `if [singleton tests fail] exit 1; else [run flux-parsl integration]` line.
